packages/flutter/lib/src/material/input_decorator.dart

```diff
@@ -477,6 +477,51 @@ enum FloatingLabelBehavior {
   always,
 }
 
+/// Defines the horizontal alignment of a floating label within an
+/// [InputDecorator].
+///
+/// See also:
+///
+///  * [InputDecoration.floatingLabelAlignment] which defines the alignment for
+///    [InputDecoration.label] or [InputDecoration.labelText].
+///  * [FloatingLabelBehavior] which defines the behaviour of the floating label.
+@immutable
+class FloatingLabelAlignment {
+  const FloatingLabelAlignment._({
+    required this.x,
+  }) : assert(x != null),
+       assert(x >= -1.0 && x <= 1.0);
+
+  /// A value ranging from -1.0 to 1.0 inclusive, where -1.0 represents the
+  /// leftmost possible position, 1.0 represents the rightmost possible
```

Comment:
Here's a reference to left/right that should probably be clarified, similar to the docs in InputDecorator.floatingLabelAlignment that explain the LTR and RTL cases.